external/loaders/loaders/mappers/_nudged/_nudged.py

```diff
@@ -207,12 +216,22 @@ def open_nudge_to_fine_multiple_datasets(
 
 
 def _get_datasets(
-    url: str, sources: Sequence[str], consolidated: bool = True
+    url: str,
+    sources: Sequence[str],
+    consolidated: bool = True,
+    cache_size_mb: float = None,
 ) -> MutableMapping[Hashable, xr.Dataset]:
     datasets: MutableMapping[Hashable, xr.Dataset] = {}
     for source in sources:
-        ds = intake.open_zarr(
-            os.path.join(url, f"{source}"), consolidated=consolidated
-        ).to_dask()
+        if cache_size_mb is not None:
+            mapper = zarr.LRUStoreCache(
+                fsspec.get_mapper(os.path.join(url, f"{source}")),
+                max_size=int(cache_size_mb * 1e6),
+            )
+            ds = xr.open_zarr(mapper, consolidated=consolidated)
+        else:
+            ds = intake.open_zarr(
+                os.path.join(url, f"{source}"), consolidated=consolidated
+            ).to_dask()
```

Comment:
Can you reduce the number of differences between these two pathways? e.g. you should be able to use the mapper returned by get_mapper directly when not using a cache, with the only change being that you wrap that mapper in a cache layer if caching is enabled. Currently the code pathways are very different, using (xarray+fsspec)/intake and using dask/maybe-non-dask(?) datasets.